src/data/adHocFilter.ts

```diff
@@ -26,37 +37,60 @@ export class AdHocFilter {
     // Semicolons are not required and cause problems when building the SQL
     sql = sql.replace(';', '');
     const ast = sqlToAST(sql);
-    const filteredAST = this.applyFiltersToAST(ast, whereClause);
-    return clausesToSql(filteredAST);
+    this.applyFiltersToAST(ast, whereClause);
+    return ASTToSql(ast);
   }
 
-  private applyFiltersToAST(ast: Map<string, Clause>, whereClause: string): Map<string, Clause> {
-    if (typeof ast.get('FROM') === 'string') {
-      const fromPhrase = ast.get('FROM')!.toString().trim();
-      const tableName = this.getTableName(fromPhrase);
-      if (tableName !== this._targetTable) {
-        return ast;
-      }
-      // If there is no defined WHERE clause create one
-      // Else add an ad hoc filter to the existing WHERE clause
-      if (ast.get('WHERE') === null) {
-        ast.set('FROM', ` ${tableName} `);
-        // set where clause to ad hoc filter and add the remaining part of the from phrase to the new WHERE phrase
-        return ast.set('WHERE', `${whereClause} ${fromPhrase.substring(tableName.length)}`);
+  private applyFiltersToAST(ast: AST, whereClause: string): AST {
```

Comment:
this doesn't look specific to adHocFilters.  could put it in ast.ts?

Reply:
That sounds reasonable to me 👍